core/src/main/scala/org/apache/spark/deploy/master/Master.scala

```diff
@@ -127,7 +128,14 @@ private[deploy] class Master(
     logInfo(s"Running Spark version ${org.apache.spark.SPARK_VERSION}")
     webUi = new MasterWebUI(this, webUiPort)
     webUi.bind()
-    masterWebUiUrl = "http://" + masterPublicAddress + ":" + webUi.boundPort
+    if (reverseProxy) {
+      masterWebUiUrl = conf.get("spark.ui.reverseProxyUrl", null)
+      if (masterWebUiUrl == null) {
+       throw new SparkException("spark.ui.reverseProxyUrl must be provided")
```

Comment:
When this is thrown the Spark Master still starts, but workings and applications can't connect to it.